src/mil_common/mil_tools/src/hardware/cpu_temp.cpp

```diff
@@ -0,0 +1,122 @@
+#include "mil_tools/hardware/cpu_temp.hpp"
+#include "mil_tools/hardware/system_info.hpp"
+#include <sensors/sensors.h>
+#include <cstring>
+#include <string>
+
+namespace mil_tools::hardware::cpu_temp
+{
+    double get_cpu_temperature()
+    {
+        // First we need to know the CPU architecture since the command/library we'll use for determining the temp differs based on the architecture
+        std::string cpu_architecture = mil_tools::hardware::system_info::get_cpu_architecture();
+
+        // If the architecture is x86, we'll use the lm-sensors/libsensors-cpp library
+        if (cpu_architecture.compare("x86_64") == 0)
+        {
+            // Initialize sensors, none-zero returned value means there was an error during initialization
+            if (sensors_init(nullptr) != 0) 
+                // Return without modifying any values
+                return -1;
+
+            // Iterate through all chips
```

Comment:
Same with some of these comments; if they are likely obvious, i would just remove them